cryptofeed/feed.py

```diff
@@ -101,12 +105,22 @@ async def book_callback(self, book: dict, book_type: str, pair: str, forced: boo
             changed, book = await self.apply_depth(book, False, pair)
             if not changed:
                 return
+
+        self.check_bid_ask_overlapping(book, pair)
         if book_type == L2_BOOK:
             await self.callback(L2_BOOK, feed=self.id, pair=pair, book=book, timestamp=timestamp, receipt_timestamp=receipt_timestamp)
         else:
             await self.callback(L3_BOOK, feed=self.id, pair=pair, book=book, timestamp=timestamp, receipt_timestamp=receipt_timestamp)
         self.updates[pair] = 0
 
+    def check_bid_ask_overlapping(self, book, pair):
+        bid, ask = book[BID], book[ASK]
+        assert isinstance(bid, sd) and isinstance(ask, sd)
```

Comment:
I'm thinking we don't need this as every exchange/feed implementation uses sorted dict.

Reply:
Removed